icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/PersonNameFormatterTest.java

```diff
@@ -0,0 +1,341 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+package com.ibm.icu.dev.test.format;
+
+import com.ibm.icu.dev.test.TestFmwk;
+import com.ibm.icu.text.PersonNameFormatter;
+import com.ibm.icu.text.SimplePersonName;
+import com.ibm.icu.util.ULocale;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.util.*;
+
+@RunWith(JUnit4.class)
+public class PersonNameFormatterTest extends TestFmwk{
+    private static class NameAndTestCases {
+        public String nameFields;
+        public String[][] testCases;
+
+        public NameAndTestCases(String nameFields, String[][] testCases) {
+            this.nameFields = nameFields;
+            this.testCases = testCases;
+        }
+    }
+
+    private void executeTestCases(NameAndTestCases[] namesAndTestCases, boolean forDebugging) {
```

Comment:
For test cases, it might be good to add the following cases:

- hu Hungarian, name with initial `ǳ` ligature: `given=ǳsárgál, surname=ǳiatkowicz`
- inital-cap: `ǲ`; allCaps: `Ǳ`, monogram: `D` or `Ǳ`?

Reply:
Done!